Formula/amp.rb

```diff
@@ -19,7 +19,9 @@ class Amp < Formula
   uses_from_macos "libiconv"
 
   def install
-    system "cargo", "install", *std_cargo_args
+    ENV.append_to_cflags "-arch #{Hardware::CPU.arch}" if Hardware::CPU.arm?
```

Comment:
Actually, now that I think about it, I'm not sure this would be the fix, since I'm pretty sure superenv already passes this flag to the compiler. Unless `cargo` gets around the shims somehow?

Reply:
Looks like it's doing precisely that for me locally, from my comment in the main body. The shim seems to be the breakage 😕.